lib.php

```diff
@@ -87,3 +87,47 @@ function auth_oidc_connectioncapability($userid, $mode = 'connect', $require = f
     }
     return $result;
 }
+
+function auth_oidc_before_footer() {
+    global $SESSION;
+    $oauth_origin = get_config('auth_oidc', 'oauthorigin');
+    $session_check_endpoint = get_config('auth_oidc', 'sessioncheckendpoint');
+    $logouturl = get_config('auth_oidc', 'logouturi');
+    $client_id = get_config('auth_oidc', 'clientid');
+    $session_state = isset($SESSION->session_state) ? $SESSION->session_state : '';
+?>
```

Comment:
there's really no need to render this check session iframe if user is not logged in. Please render the iframe only if user is recognized as authenticated user.